### PR TITLE
TRT-588 - Add variable subsetting option to HyBIG.

### PR DESCRIPTION
**Pull request by owenlittlejohns**
## Jira Issue ID

TRT-588

## Description

This PR updates HyBIG to allow for variable subsetting (only one variable). This is to support browse image generation for PREFIRE, which requires a custom colour map, currently specified via a UMM-Var record.

## Local Test Steps

* Pull this branch.
* Pull [this branch](https://github.com/nasa/harmony-regression-tests/tree/TRT-588-HyBIG-variable) for the regression tests. (Unless [this PR](https://github.com/nasa/harmony-regression-tests/pull/105) is already merged - if so `main` will work)
* Stand up a local Harmony instance with [the latest HyBIG image](https://github.com/nasa/harmony-browse-image-generator/pkgs/container/harmony-browse-image-generator) (there are no actual changes to HyBIG).
* Run the updated HyBIG regression test notebook against your local Harmony instance. It should all pass, including the PREFIRE test at the end.
* Visually inspect the tiles. Those with data going through them should have data that has a non-greyscale colour map (see example screenshot attached).

## PR Acceptance Checklist
* [x] Acceptance criteria met
* ~~Tests added/updated (if needed) and~~ passing
* ~~Documentation updated (if needed)~~

Screenshot 2024-10-16 at 2 41 19 PM

### CODE REVIEW

**Comment by owenlittlejohns** [comment]
@hazemmahmoud88 - just for your awareness.

**Review by flamingbear — APPROVED** [review]
This looks like what I expected.  
I ran the regressions on the existing harmony and got the error:

``` 
Exception: ('Unprocessable Entity', 'Error: the requested combination of operations: variable subsetting on C1263096190-EEDTEST is unsupported')
```

Which is what I would expect before this change.  I pulled this branch and I was able to restart my Harmony-In-A-Box and run the regression tests without failure (after getting on the VPN). 

I downloaded the generated results and was able to load color tiles into QGIS.  Looks great!


Screenshot 2024-10-17 at 8 45 36 AM

File: config/services.yml [hunk 1]
```diff
@@ -397,6 +397,8 @@ https://cmr.earthdata.nasa.gov:
         bbox: false
         shape: false
         temporal: false
+        variable: true
+        multiple_variable: false
```
Comment (chris-durbin) [inline config/services.yml:401]: Just FYI that we don't do anything with this information yet. I can write a ticket to make it so services don't match to a request if they request multiple variables and the service supports variable, but not multiple variable.

**Review by chris-durbin — APPROVED** [review]
Reminder to update the production UMM-S record after this has been deployed to prod:

```
ERROR: Validation failures for sds/HyBIG and S2697183066-XYZ_PROV:
    - Variable subset mismatch: harmony is true and UMM-S is false.
```

**Pull request merged**